### ISSUE TITLE
OpenCL not activated darktable 4.0 macos M1 mackbook air

### ISSUE THREAD

**Issue by seekjimmyg** [darktable-org/darktable#12119]
**Describe the bug/issue**
Opencl is not activated (greyed out) on darktable 4.0 running on Macbook air M1. Opencl was activated in darktable 3.8 on the same laptop.

**Expected behavior**
Opencl options not greyed out

**Screenshots**

<img width="560" alt="Screen Shot 2022-07-03 at 11 50 37 AM" src="https://user-images.githubusercontent.com/58573841/177049439-87f082f3-f9f8-420e-8961-b828756c7dd4.png">

output for darktable -d opencl for both 3.8 and 4 attached

[darktable3-8-opencl.txt](https://github.com/darktable-org/darktable/files/9035166/darktable3-8-opencl.txt)
[darktable4-0-opencl.txt](https://github.com/darktable-org/darktable/files/9035167/darktable4-0-opencl.txt)

**Comment by jenshannoschwalm** [comment]
Hi, as mentioned on pixls.up there is the page in the manual
https://darktable-org.github.io/dtdocs/en/special-topics/mem-performance/

You should be able to open your darktablerc file and locate the line starting with `cldevice_v4_` -- on my system it's 

`cldevice_v4_quadrortx4000=0 250 0 16 16 1024 0 0 0.017853`

I guess on your system the 8th number is 1 (means disabled) - set it to 0 please.

Can you confirm this?

Nevertheless this is not supposed to happen, i will take care of it. Sorry about it

**Comment by joseph-holland** [comment]
Hi @jenshannoschwalm,

I can confirm on my M1 Macbook Air that the line now appears like:
```cldevice_v4_applem1=0 1000 0 16 16 128 0 1 0.000000```

Setting the 8th number to 0 seems to have worked, although my changes seem to have gotten overwritten the first time and I had to edit the darktablerc file again and quit and reopen darktable.

Now I can see that activate OpenCL support is no longer greyed out and it now checked.

Thanks

**Comment by ptilopteri** [comment]
fwiw:  My i7 with GeForce RTX 3060 Ti and darktablerc as installed
containing "tuneopencl=FALSE" also grays out the ability to configure 
opencl with 4.1.0+22~gd39d5f6.  indeed darktable-clinfo shows opencl 
as not being available.

driver: nvidia v: 515.57

note that changing tuenopencl=TRUE enables configuration of opencl
and after running darktable once, darktable-clinfo shows opencl as available
and active.

THIS SHOULD NOT BE.

setting tuneopencl should not preclude the ability of darktable to utilize 
opencl.

**Comment by jenshannoschwalm** [comment]
This is not related. Dt simply can not change what clinfo reports, no way.

If the device was not available for some reason dt can't enable CL. Did you restart your system?

**Comment by ptilopteri** [comment]
no, I did not restart the system after editing "tuneopencl=FALSE" to TRUE.
prior to the noted edit, darktable-clinfo reported opencl no available
after the noted edit, darktable-clinfo reports opencl available and used.

the default "tuneoencl" setting caused darktable to fail to use opencl.

**Comment by seekjimmyg** [comment]
@jenshannoschwalm your fix worked. Changing the 8th # to 0 worked. Thank you very much.

**Comment by jenshannoschwalm** [comment]
@seekjimmyg please concider to close the issue for now

**Comment by dmitry-kabanov** [comment]
> Hi, as mentioned on pixls.up there is the page in the manual https://darktable-org.github.io/dtdocs/en/special-topics/mem-performance/
> 
> You should be able to open your darktablerc file and locate the line starting with `cldevice_v4_` -- on my system it's
> 
> `cldevice_v4_quadrortx4000=0 250 0 16 16 1024 0 0 0.017853`
> 
> I guess on your system the 8th number is 1 (means disabled) - set it to 0 please.
> 
> Can you confirm this?
> 
> Nevertheless this is not supposed to happen, i will take care of it. Sorry about it

I had the same problem on MacBook Pro 13 M1 2020, and the proposed change from 1 to 0 helped. Then darktable compiled all kernels and initialized with OpenCL!
Thank you, @jenshannoschwalm!

**Comment by apfelkraut** [comment]
Just want to share my experiences with OpenCl on macOS 12.X, M1, MacbookAir. 

After updating to darktable 4.0 and starting for the first time (after the update), OpenCL was not available (greyed out) as reported by @seekjimmyg. Sorry, I did not check for any details.

A simple restart of darktable 4.0 made it work for me and it is available since then. 

I remember a previous issue stating similar problems: https://github.com/darktable-org/darktable/issues/2810

In short: Compiling kernels for OpenCl fail during initial start. Due to that OpenCl is not available. Restarting darktable, at worst several times, does the trick though the actual root cause is unknown.
